doc/learn/syntax-trees-with-typescript.md

```diff
@@ -0,0 +1,277 @@
+`Link`, and many more.
+A full list of nodes can be found in the [specification](https://github.com/syntax-tree/mdast#readme).
+The types are available in a types only package: [`@types/mdast`](https://www.npmjs.com/package/@types/mdast).
+
+Install:
+
+```bash
+npm install --save-dev @types/mdast
+```
+
+To import the types into a TypeScript file, use:
+
+```ts
+import type {Heading, Code, Link} from 'mdast'
+```
+
+To import the types in [JSDoc TypeScript](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html), use:
+
+```js
+/**
+ * @typedef {import('mdast').Heading} Heading
+ * @typedef {import('mdast').Code} Code
+ * @typedef {import('mdast').Link} Link
+ */
+```
+
+### HAST (HTML)
+
+[hast (**h**ypertext **a**bstract **s**yntax **t**ree)](https://github.com/syntax-tree/hast#readme)
+extends unist with types specific for HTML such as `Element`, `Comment`,
+`DocType`, and many more.
+A full list of nodes can be found in the [specification](https://github.com/syntax-tree/hast#readme).
+The types are available in a types only package: [`@types/hast`](https://www.npmjs.com/package/@types/hast).
+
+Install:
+
+```bash
+npm install --save-dev @types/hast
+```
+
+To import the types into a TypeScript file, use:
+
+```ts
+import type {Element, Comment, DocType} from 'hast'
+```
+
+To import the types in [JSDoc TypeScript](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html), use:
+
+```js
+/**
+ * @typedef {import('hast').Element} Element
+ * @typedef {import('hast').Comment} Comment
+ * @typedef {import('hast').DocType} DocType
+ */
+```
+
+### XAST (XML)
+
+[xast (e**x**tensible **a**bstract **s**yntax **t**ree)](https://github.com/syntax-tree/xast#readme)
+extends unist with types specific for HTML such as `Element`, `CData`,
+`Instruction`, and many more.
+A full list of nodes can be found in the [specification](https://github.com/syntax-tree/xast#readme).
+The types are available in a types only package: [`@types/xast`](https://www.npmjs.com/package/@types/xast).
+
+Install:
+
+```bash
+npm install --save-dev @types/xast
+```
+
+To import the types into a TypeScript file, use:
+
+```ts
+import type {Element, CData, Instruction} from 'xast'
+```
+
+To import the types in [JSDoc TypeScript](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html), use:
+
+```js
+/**
+ * @typedef {import('xast').Element} Element
+ * @typedef {import('xast').CData} CData
+ * @typedef {import('xast').Instruction} Instruction
+ */
+```
+
+### Summary
+
+*   Unified provides types for each language’s syntax tree
+*   These types can be import into TypeScript projects and into JSDoc projects
+
+### Next steps
+
+*   [Learn to traverse syntax trees with TypeScript](/learn/recipe/tree-traversal-typescript)
+*   [Learn to narrow `Node` to a more specific type with TypeScript](/learn/recipe/node-type-narrowing-in-typescript)
+*   [Learn to build content with syntax trees in TypeScript](/learn/recipe/build-a-syntax-tree-typescript)
```

Comment:
I think these need final slashes (although GH and most servers redirect)

Reply:
added